src/Forms/RequiredFields.php

```diff
@@ -104,16 +104,29 @@ public function php($data)
                 $fieldName = $fieldName->getName();
             } else {
                 $formField = $fields->dataFieldByName($fieldName);
+                if (is_null($formField)) {
+                    $formField = $fields->dataFieldByName($fieldName . 'ID');
+                }
```

Comment:
If the form field doesn't implement `HasOneRelationFieldInterface`, we should set the variable back to `null` - this logic should only be done for `has_one` fields.

Reply:
Have removed code